keysign/app.py

```diff
@@ -260,50 +265,55 @@ def on_keylist_mapped(self, keylistwidget):
             Gtk.IconSize.BUTTON))
         # We don't support refreshing for now.
         self.header_button.set_sensitive(False)
-
+        self.internet_toggle.show()
 
     def on_send_stack_mapped(self, stack):
         log.debug("send stack becomes visible!")
+        # Adjust the top bar buttons
+        self.on_send_stack_switch(stack)
 
     def on_keypresent_mapped(self, kpw):
         log.debug("keypresent becomes visible!")
         self.header_button.set_sensitive(True)
         self.header_button.set_image(
             Gtk.Image.new_from_icon_name("go-previous",
             Gtk.IconSize.BUTTON))
+        self.internet_toggle.hide()
 
     def on_scanner_mapped(self, scanner):
         log.debug("scanner becomes visible!")
         self.header_button.set_sensitive(False)
         self.header_button.set_image(
             Gtk.Image.new_from_icon_name("go-previous",
             Gtk.IconSize.BUTTON))
+        self.internet_toggle.hide()
 
     def on_presign_mapped(self, psw):
         log.debug("presign becomes visible!")
         self.header_button.set_sensitive(True)
         self.header_button.set_image(
             Gtk.Image.new_from_icon_name("go-previous",
             Gtk.IconSize.BUTTON))
+        self.internet_toggle.hide()
 
 
-
-
-def main(args = []):
+def main(args=[]):
     logging.basicConfig(
         level=logging.DEBUG,
         format='%(name)s (%(levelname)s): %(message)s')
     log = logging.getLogger(__name__)
     log.debug('Running main with args: %s', args)
     if not args:
         args = []
-    Gst.init()
+    Gst.init(None)
 
     app = KeysignApp()
-    try:
-        GLib.unix_signal_add_full(GLib.PRIORITY_HIGH, signal.SIGINT, lambda *args : app.quit(), None)
-    except AttributeError:
-        pass
+
+    def stop(signum, stackframe):
+        app.quit()
+        reactor.callFromThread(reactor.stop)
+
+    signal.signal(signal.SIGINT, stop)
```

Comment:
I believe the reason for the try/except trickery was that SIGINT does not necessarily (on old Windows systems or so...).  Would `GLib.unix_signal_add_full(GLib.PRIORITY_HIGH, signal.SIGINT, lambda *args : (app.quit(),  reactor.callFromThread(reactor.stop)), None)` work? Then it's not your fault if things break later ;-)